apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs

```diff
@@ -289,6 +292,23 @@ private LogEntry BuildLogEntry(LoggingEvent loggingEvent)
             return logEntry;
         }
 
+        private void TryAddGitRevisionId(Dictionary<string, string> labels)
+        {
+            string gitId = null;
+            try
+            {
+                gitId = SourceContext.AppSourceContext?.Git?.RevisionId;
+            }
+            catch (Exception ex) when (ex is System.Security.SecurityException || ex is InvalidProtocolBufferException
+                || ex is InvalidJsonException || ex is UnauthorizedAccessException)
+            { }
+
+            if (!String.IsNullOrWhiteSpace(gitId))
```

Comment:
I'd move this inside the catch block - then you don't need the variable to be declared first. The Add call won't throw any of the caught exceptions, so it won't change behaviour.